operator/opensearchcontroller/observe.go

```diff
@@ -0,0 +1,147 @@
+package opensearchcontroller
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+	exoscaleapi "github.com/exoscale/egoscale/v2/api"
+	"github.com/exoscale/egoscale/v2/oapi"
+	"github.com/go-logr/logr"
+	exoscalev1 "github.com/vshn/provider-exoscale/apis/exoscale/v1"
+	"github.com/vshn/provider-exoscale/operator/mapper"
+	"k8s.io/utils/pointer"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+)
+
+// Observe implements managed.ExternalClient.
+func (p *pipeline) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) {
+	log := controllerruntime.LoggerFrom(ctx)
+	log.V(1).Info("observing resource")
+
+	openSearchInstance := mg.(*exoscalev1.OpenSearch)
+
+	resp, err := p.exo.GetDbaasServiceOpensearchWithResponse(ctx, oapi.DbaasServiceName(openSearchInstance.GetInstanceName()))
+	if err != nil {
+		if errors.Is(err, exoscaleapi.ErrNotFound) {
+			return managed.ExternalObservation{ResourceExists: false}, nil
```

Comment:
nit: `ResourceExists: false` is not necessary since false is the default. Doesn't hurt either though ;)